src/mesh/forest/tree.cpp

```diff
@@ -43,16 +43,21 @@ Tree::Tree(Tree::private_t, std::int64_t id, int ndim, int root_level, RegionSiz
     for (int k = 0; k < (ndim > 2 ? (1LL << l) : 1); ++k) {
       for (int j = 0; j < (ndim > 1 ? (1LL << l) : 1); ++j) {
         for (int i = 0; i < (ndim > 0 ? (1LL << l) : 1); ++i) {
+          LogicalLocation loc(my_id, l, i, j, k);
           if (l == root_level) {
-            leaves.emplace(std::make_pair(LogicalLocation(my_id, l, i, j, k),
-                                          std::make_pair(-1, -1)));
+            leaves.emplace(LocMapEntry(loc, -1, -1));
           } else {
-            internal_nodes.emplace(my_id, l, i, j, k);
+            internal_nodes.emplace(LocMapEntry(loc, -1, -1));
           }
         }
       }
     }
   }
+
+  // Build in negative levels
+  for (int l = -20; l < 0; ++l) {
```

Comment:
Is -20 just a large enough number that we should never reach it?

Reply:
yeah, that would imply we had a meshblock size of (2^20)^D.